src/components/CampaignPage.js

```diff
@@ -168,21 +189,53 @@ class CampaignPage extends Component {
                     showModal: true, modalTitle: 'complete',
                     modalMessage: 'thankYouDonation',
                     errorIcon: 'CheckCircle', modalButtonMessage: 'closeBtn',
-                    modalButtonVariant: '#588157', waitToClose: false
+                    modalButtonVariant: '#588157', waitToClose: false, tryAgainCC: false, ccinfo: {}
                 });
             } else {
                 this.setState({
                     showModal: true, modalTitle: 'failed', modalMessage: PAYMENT_ERROR_MESSAGES[resp.data.paymentStatus],
-                    errorIcon: 'XCircle', modalButtonMessage: 'closeBtn',
-                    modalButtonVariant: '#E63C36', waitToClose: false
+                    errorIcon: 'XCircle', modalButtonMessage: 'tryAgain',
+                    modalButtonVariant: '#E63C36', waitToClose: false, tryAgainCC: true
                 });
+                this.setState(prevState => ({
+                    ccinfo: {
+                        ...prevState.ccinfo,
+                        ccError : PAYMENT_ERROR_MESSAGES[resp.data.paymentStatus]
+                    }
+                }));
             }
         } catch (err) {
+            console.log(err.response.data);
+            let errorFound = false;
+            Object.keys(CC_INFO_FIELDS_ERRORS).every((key)=>{
+                if(err.response.data.paymentStatus.message.includes(key)){
+                    this.setState({modalMessage: CC_INFO_FIELDS_ERRORS[key]});
+                    this.setState(prevState => ({
+                        ccinfo: {
+                            ...prevState.ccinfo,
+                            ccError: CC_INFO_FIELDS_ERRORS[key],
+                            ccErrorType: `${key}Input` 
+                        }
+                    }));
+                    errorFound = true;
+                    return false;
```

Comment:
what is the meaning of returning false here and returning true two lines down? How is the return value used?

Reply:
.every() loop is like a foreach loop but u can break out of it by returning false. Or stay in by returning true. Once the error is found, no reason to keep looping